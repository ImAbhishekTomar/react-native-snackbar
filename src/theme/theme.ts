import * as React from 'react';

export type ThemeProps = {
    textColor?: string | number;
    backgroundColor?: string;
    fontFamily?: string;
}

export type SnackbarThemeType = {
    success?: ThemeProps;
    warning?: ThemeProps;
    danger?: ThemeProps;
    default?: ThemeProps;
    dark?: ThemeProps;
    light?: ThemeProps;
}

export const SnackbarDefaultTheme: SnackbarThemeType = {
    success: {
        backgroundColor: '#92E6A6',
        textColor: '#007725'
    },
    warning: {
        backgroundColor: '#F8D953',
        textColor: '#795400'
    },
    danger: {
        backgroundColor: '#E69292',
        textColor: '#770009'
    },
    default: {
        backgroundColor: '#4D4D4D',
        textColor: '#fff'
    },
    dark: {
        backgroundColor: '#4D4D4D',
        textColor: '#fff'
    },
    light: {
        backgroundColor: '#fcfcfc',
        textColor: '#000'
    },
};

//for theme Context support -- In development
const ContextSnackbarTheme = React.createContext(SnackbarDefaultTheme);

const SnackbarThemeSetting = (personalizeTheme: SnackbarThemeType = SnackbarDefaultTheme): SnackbarThemeType => {
    return personalizeTheme;
};

export const SNACKBARTYPE_SUCCESS = SnackbarThemeSetting().success;
export const SNACKBARTYPE_WARNING = SnackbarThemeSetting().warning;
export const SNACKBARTYPE_DANGER = SnackbarThemeSetting().danger;
export const SNACKBARTYPE_DEFAULT = SnackbarThemeSetting().default;

//For dark and light support
export const SNACKBARTYPE_DARK = SnackbarThemeSetting().dark;
export const SNACKBARTYPE_LIGHT = SnackbarThemeSetting().light;


//export default SnackbarThemeSetting;


