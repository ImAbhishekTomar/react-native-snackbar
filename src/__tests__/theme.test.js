import { SNACKBARTYPE_DANGER } from '../theme/theme';
import { NativeModules } from 'react-native';


describe('Snackbar module', () => {
    let Snackbar;

    beforeAll(() => {
        NativeModules.RNSnackbar = {
            show: jest.fn(),
        };

        // Require Snackbar only after NativeModules have been mocked.
        // eslint-disable-next-line global-require
        Snackbar = require('../index').default;
    });

    beforeEach(() => {
        NativeModules.RNSnackbar.show.mockClear();
    });

    //INFO: #TEST 1
    it('#1-check full setting with theme', () => {
        Snackbar.show({
            text: 'Hello world',
            duration: 0,
            ...SNACKBARTYPE_DANGER,
            action: {
                text: 'UNDO',
                textColor: 'green',
                onPress: jest.fn(),
            },
        });
        expect(NativeModules.RNSnackbar.show.mock.calls).toMatchSnapshot();
    });

});