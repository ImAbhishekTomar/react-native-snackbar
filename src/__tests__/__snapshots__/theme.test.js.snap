// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar module #1-check full setting with theme 1`] = `
Array [
  Array [
    Object {
      "action": Object {
        "onPress": [MockFunction],
        "text": "UNDO",
        "textColor": 4278222848,
      },
      "backgroundColor": 4293300882,
      "duration": 0,
      "text": "Hello world",
      "textColor": 4285988873,
    },
    [MockFunction],
  ],
]
`;
